File: app/src/main/java/ua/meugen/android/pdfcomposer/ui/activities/createnew/fragments/createnew/presenter/CreateNewPresenter.java
package ua.meugen.android.pdfcomposer.ui.activities.createnew.fragments.createnew.presenter;

import ua.meugen.android.pdfcomposer.ui.activities.base.fragment.presenter.MvpPresenter;
import ua.meugen.android.pdfcomposer.ui.activities.createnew.fragments.createnew.state.CreateNewState;


public interface CreateNewPresenter extends MvpPresenter<CreateNewState> {

    void setup();

    void onImageSelected(String imageUri);

    void onTextChanged(String text);

    void addPage();

    void exportPdf();
}

File: app/src/main/java/ua/meugen/android/pdfcomposer/ui/activities/createnew/fragments/createnew/presenter/CreateNewPresenter.kt
package ua.meugen.android.pdfcomposer.ui.activities.createnew.fragments.createnew.presenter

import ua.meugen.android.pdfcomposer.ui.activities.base.fragment.presenter.MvpPresenter
import ua.meugen.android.pdfcomposer.ui.activities.createnew.fragments.createnew.state.CreateNewState


interface CreateNewPresenter : MvpPresenter<CreateNewState> {

    fun setup()

    fun onImageSelected(imageUri: String)

    fun onTextChanged(text: String)

    fun addPage()

    fun exportPdf()
}
